controllers/challenge.js

```diff
@@ -12,10 +12,17 @@ const sendChallengeResponse = async (req, res) => {
   try {
     if (req.method === 'GET') {
       const allChallenges = await challengeQuery.fetchChallenges()
-      if (allChallenges.length > 0) {
+      const newChallenges = await Promise.all(allChallenges.map(async (challenge) => {
+        const getParticipants = await challengeQuery.fetchParticipants(challenge.participants)
```

Comment:
A lot is happening here, maybe you could use one more variable to keep things clear?